src/uct/tcp/tcp_ep.c

```diff
@@ -233,15 +233,14 @@ ucs_status_t uct_tcp_ep_add_ctx_cap(uct_tcp_ep_t *ep,
 {
     uct_tcp_iface_t *iface = ucs_derived_of(ep->super.super.iface,
                                             uct_tcp_iface_t);
-    uint8_t prev_caps      = ep->ctx_caps;
+    uint8_t prev_caps      = ep->ctx_caps & UCT_TCP_EP_CTX_CAPS;
 
     uct_tcp_ep_change_ctx_caps(ep, ep->ctx_caps | UCS_BIT(cap));
-    if (!uct_tcp_ep_is_self(ep) && (prev_caps != ep->ctx_caps)) {
+    if (!uct_tcp_ep_is_self(ep) &&
+        (prev_caps != (ep->ctx_caps & UCT_TCP_EP_CTX_CAPS))) {
```

Comment:
i don't think we need this: can assert that 'cap' is one of TX,RX (which is expected).
Then, adding cap will not change unrelated flags. so no need to do ` & UCT_TCP_EP_CTX_CAPS` in here and in line 236

Reply:
then we need to do it at the 241 line